### DISCUSSION TITLE
Graduate Thesis Project

### DISCUSSION THREAD

**Discussion by Duxtrous** [BHoM/BHoM#1454] [General]
Hello all!

I am a graduate student studying structural design at an Architectural Engineering school in the US. I have decided that for my thesis I wanted to do something related to the calculations of embodied carbon and landed on utilizing Dynamo to automate this process. The BHoM caught my eye as I saw many articles utilizing it on Autodesk University and it seemed very user friendly. I have minimal coding experience but always greatly enjoyed coding and typically had to help my fellow engineers through their coding courses.

I have had various issues pop up in the Dynamo program and cannot easily find locations for trouble shooting. I am realizing that this discussion board is the most optimal place to get help, but I was wondering if I could even take it a step further by directly getting in touch with a developer of the program. I know this is a large ask as many people here work on this code in their freetime but I would love to be able to understand this program on a deeper level so that I may be able to spread it to other's in my field. Automation and coding are not heavily pushed in the world of architectural engineering in my area, but I know it's the puzzle piece that the field is missing. I plan to make videos walking through my processes, not just for my thesis, but also to be shared on YouTube so that others may follow in my footsteps. Having a source to be able to consult on issues directly and potentially even be able to video call to work through problems would be a godsend as I find walking through solutions without visuals can often be difficult.

If this is not an appropriate post for this forum that is completely understandable, please just delete my post and I will upload problem specific posts in the future.

Thank you for your time and work towards open-source programming!
- Dux

**Comment by FraserGreenroyd** [comment]
Hi @Duxtrous , welcome to the BHoM!

We do have a [Dynamo_UI](https://github.com/BHoM/Dynamo_UI) which comes packaged with the quarterly installers, however, we've encountered numerous issues with building for Dynamo that we've stopped supporting it for all but compilation at this stage, focusing instead on the information we can pull using [Revit_Toolkit](https://github.com/BHoM/Revit_Toolkit) and analyse with the [LifeCycleAssessment_Toolkit](https://github.com/BHoM/LifeCycleAssessment_Toolkit) in [Grasshopper](https://github.com/BHoM/Grasshopper_UI) and [Excel](https://github.com/BHoM/Excel_UI).

@michaelhoehn is the primary person to speak to with regard to the analysis of embodied carbon though there are many others who may be able to support as well, but @michaelhoehn is US based so is closer to your timezone than the UK/EU based contributors :smile:

I'm happy to have a call/chat about the UI strategy and how the code reflects up into Dynamo/the other UIs as well if that would be beneficial, but I suspect the problems that are popping up are going to be difficult to resolve with the resources available - but feel free to reach out if you want a call, my email is on my profile :smile:

**Reply by Duxtrous** [reply]
@FraserGreenroyd ,

Thank you so much for the swift reply! I did notice that a lot of the newer updates seemed to have been for Grasshopper and Rhino so I will probably just switch over to that in the coming week and purchase a license if my project ends up taking longer than the 90 day trial. This makes a lot of sense as sometimes processes would work perfectly fine until suddenly I ran the program and bam, now it shot back run.caller errors. 

I will definitely be reaching out once I get my project set up in Grasshopper. Timezone isn't too much of an issue for me as university destroyed my consistant routine long ago. I'll be sure to shoot michael a message too once I get all of the pull commands back up and running.


Thanks again!
- Dux

**Reply by FraserGreenroyd** [reply]
That sounds great @Duxtrous 😄 look forward to hearing how it goes and also have a strange routine so presumably should be easy to sort out a meeting when you're ready 😄
